src/server/routes/carts.js

```diff
@@ -1,12 +1,17 @@
-
 import express from "express";
-export function createCartsRouter({ cartRepo }) {
+export function createCartsRouter({ cartRepo, userRepo }) {
     const router = express.Router();
 
     // get cart by userId
     router.get("/:userId", async (req, res) => {
         const userId = parseInt(req.params.userId);
+        //check if  the user actually exists
+        if (!userRepo.getUserById/userId){
```

Comment:
If the user is not found, the cart for the user can't be found anyway. So, is this needed?